src/main/java/fr/openent/competences/controllers/CompetenceNoteController.java

```diff
@@ -462,34 +440,37 @@ public void deleteCompetencesNotesDevoir(final HttpServerRequest request) {
         competencesNotesService.dropCompetencesNotesDevoir(oIdsJsonArray, arrayResponseHandler(request));
     }
 
-    private void callGetCompetenceNote(String idClasse, Long idPeriode, Integer typeClasse, List<String> idDomaines,
+    private void callGetCompetenceNote(String idClasse, Long idPeriode, Integer typeClasse, String idStructure,
                                        final HttpServerRequest request) {
-        new DefaultUtilsService(this.eb).studentIdAvailableForPeriode(idClasse, idPeriode, typeClasse,
-                new Handler<Either<String, JsonArray>>() {
-                    @Override
-                    public void handle(Either<String, JsonArray> event) {
-                        if (event.isRight()) {
-                            JsonArray queryResult = event.right().getValue();
-                            List<String> idEleves = new ArrayList<String>();
-
-                            if (queryResult != null) {
-                                for (int i = 0; i < queryResult.size(); i++) {
-                                    idEleves.add(queryResult.getString(i));
-                                }
-                            }
-                            if (idDomaines != null) {
-                                callCompetencesNotesDomaineService(idEleves, idPeriode, idDomaines, request);
-                            } else {
-                                callCompetencesNotesService(idEleves, idPeriode, request);
-                            }
+        new DefaultUtilsService(this.eb).studentIdAvailableForPeriode(idClasse, idPeriode, typeClasse, event -> {
+            if (event.isRight()) {
+                JsonArray queryResult = event.right().getValue();
+                List<String> idEleves = new ArrayList<String>();
```

Comment:
Pas besoin de préciser le typage de ton ArrayList

```suggestion
                List<String> idEleves = new ArrayList<>();
```
source: https://sonarsource.atlassian.net/browse/RSPEC-2293#:~:text=Java%207%20introduced%20the%20diamond,compiler%20will%20infer%20the%20type.